controllers/client_policy.go

```diff
@@ -206,28 +226,50 @@ func (r *SingleClusterReconciler) appendCACertFromSecret(
 
 	found := &corev1.Secret{}
 
-	secretName := namespacedSecret(secretSource, defaultNamespace)
-	if err := r.Client.Get(context.TODO(), secretName, found); err != nil {
-		r.Log.Error(
-			err,
-			"Failed to get secret certificates to the pool, returning empty certPool",
-			"secret", secretName,
-		)
+	if secretSource.CaCertPath != nil {
+		secretName := namespacedSecret(secretSource.CaCertPath.SecretNamespace,
+			secretSource.CaCertPath.SecretName, defaultNamespace)
+		if err := r.Client.Get(context.TODO(), secretName, found); err != nil {
+			r.Log.Error(
+				err,
+				"Failed to get CA certificates secret, returning empty certPool",
+				"secret", secretName,
+			)
 
-		return serverPool
-	}
+			return serverPool
+		}
 
-	if caData, ok := found.Data[secretSource.CaCertsFilename]; ok {
-		r.Log.V(1).Info(
-			"Adding cert to tls server-pool from the secret.", "secret",
-			secretName,
-		)
-		serverPool.AppendCertsFromPEM(caData)
+		for file, caData := range found.Data {
+			r.Log.V(1).Info(
+				"Adding cert to tls server-pool from the secret.", "secret",
+				secretName, "file", file,
+			)
+			serverPool.AppendCertsFromPEM(caData)
+		}
 	} else {
-		r.Log.V(1).Info(
-			"WARN: Can't find ca-file in the secret. using default certPool.",
-			"secret", secretName, "ca-file", secretSource.CaCertsFilename,
-		)
+		secretName := namespacedSecret(secretSource.SecretNamespace, secretSource.SecretName, defaultNamespace)
```

Comment:
if ca-file == "", we can simply go out.

Reply:
why to check tis.ca-file, here we are fetching certs from operatorClientCert.secretCertSource?
Are you saying to skip loading CaCertsFilename if tis.ca-file is not set and same with ca-path?